doc/source/dev/newfeatures.rst

```diff
@@ -0,0 +1,40 @@
+Deciding on new features
+========================
+The general decision rule to accept a proposed new feature has so far
+been generally conditional on:
+
+1. The method is applicable in many fields and "generally agreed" to
+   be useful,
+2. Fits the topic of the submodule, and does not require extensive
+   support frameworks to operate,
+3. The implementation looks sound and unlikely to need much tweaking in
+   the future, and
+4. Someone wants to do it.
+
+Although it's difficult to give hard rules on what "generally useful
+and generally agreed to work" means, it may help to weigh the following
+against each other:
+
+- Is the method used/useful in different domains in practice?
+  How much domain-specific background knowledge is needed to use it
+  properly?
+- Consider the code already in the module.  Is what you are adding
+  an omission?  Does it solve a problem that you'd expect the module
+  be able to solve?  Does it supplement an existing feature in
+  a significant way?
+- Consider the equivalence class of similar methods / features usually
+  expected. Among them, what would in principle be the minimal set so
+  that there's not a glaring omission in the offered features remaining?
+  How much stuff would that be? Does including a representative one of
+  them cover most use cases? Would it in principle sound reasonable to
+  include everything from the minimal set in the module?
```

Comment:
Not sure about that one. I think in the past we've been designing similar APIs to Matlab/Octave because those were familiar to a lot of people, but we've stopped doing that quite a while ago I think and now just choose based on what makes sense.